jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -455,6 +455,10 @@ private void extractQueryParameters()
     /* ------------------------------------------------------------ */
     private void extractContentParameters()
     {
+        // Content cannot be encoded
+        if (_metaData!=null && getHttpFields().contains(HttpHeader.CONTENT_ENCODING))
+            throw new BadMessageException(HttpStatus.NOT_IMPLEMENTED_501,"Encoded Content");
```

Comment:
This is a confusing error message to a user.  

In a `application/x-www-form-urlencoding` of course the content is encoded, why would jetty not support that?

Perhaps it should say ...

``` java
throw new BadMessageException(HttpStatus.NOT_IMPLEMENTED_501, 
    "Unsupported Content-Encoding: " + contentEncodingValue);
```